stdpopsim/slim_engine.py

```diff
@@ -848,13 +848,31 @@ def matrix2str(
 
     # Mutation types.
     for i, m in enumerate(mutation_types):
-        distrib_args = [str(arg) for arg in m.distribution_args]
-        distrib_args[m.Q_scaled_index] = "Q * " + distrib_args[m.Q_scaled_index]
-        distrib_args = ", ".join(distrib_args)
-        printsc(
-            f"    initializeMutationType({i}, {m.dominance_coeff}, "
-            + f'"{m.distribution_type}", {distrib_args});'
-        )
+        if m.Q_scaled_index[0] is not None:
+            distrib_args = [str(arg) for arg in m.distribution_args]
+            if len(m.Q_scaled_index) == 1:
+                distrib_args[m.Q_scaled_index[0]] = (
+                    "Q * " + distrib_args[m.Q_scaled_index[0]]
+                )
+                distrib_args = ", ".join(distrib_args)
+            # dealing with normal distribution (Q * by both mean and sd):
+            else:
+                for i in m.Q_scaled_index:
+                    distrib_args[m.Q_scaled_index[i]] = (
+                        "Q * " + distrib_args[m.Q_scaled_index[i]]
+                    )
+                distrib_args = ", ".join(distrib_args)
+            printsc(
+                f"    initializeMutationType({i}, {m.dominance_coeff}, "
+                + f'"{m.distribution_type}", {distrib_args});'
+            )
+        # dealing with distributions given by "s" Eidos script
+        else:
+            distrib_args = m.distribution_args
+            printsc(
+                f"    initializeMutationType({i}, {m.dominance_coeff}, "
+                + f'"{m.distribution_type}", {distrib_args});'
+            )
```

Comment:
this looks like it will work - but, I think it can be simpler - if we change it to `"s": []` above, then I think this entire bit can be just your inner loop:
```
             distrib_args = [str(arg) for arg in m.distribution_args]
             for i in m.Q_scaled_index:
                    distrib_args[m.Q_scaled_index[i]] = (
                        "Q * " + distrib_args[m.Q_scaled_index[i]]
                    )
             distrib_args = ", ".join(distrib_args)
             printsc(
                f"    initializeMutationType({i}, {m.dominance_coeff}, "
                + f'"{m.distribution_type}", {distrib_args});'
             )
```
... am I missing something?